crates/runtime/src/lib.rs

```diff
@@ -443,22 +443,53 @@ impl Runtime {
     pub async fn load_components(&self) {
         self.start_extensions().await;
 
-        self.load_embeddings().await; // Must be loaded before datasets
-
-        let mut futures: Vec<Pin<Box<dyn Future<Output = ()>>>> = vec![
-            Box::pin(async {
-                if let Err(err) = self.init_task_history().await {
+        // Must be loaded before datasets
+        self.load_embeddings().await;
+
+        // Spawn each component load in its own task to run in parallel
+        let task_history = tokio::spawn({
+            let self_clone = self.clone();
+            async move {
+                if let Err(err) = self_clone.init_task_history().await {
                     tracing::warn!("Creating internal task history table: {err}");
-                };
-            }),
-            Box::pin(self.init_results_cache()),
-            Box::pin(self.load_datasets()),
-            Box::pin(self.load_catalogs()),
-        ];
+                }
+            }
+        });
 
-        futures.push(Box::pin(self.load_models()));
+        let results_cache = tokio::spawn({
+            let self_clone = self.clone();
+            async move {
+                self_clone.init_results_cache().await;
+            }
+        });
+
+        let datasets = tokio::spawn({
+            let self_clone = self.clone();
+            async move {
+                self_clone.load_datasets().await;
+            }
+        });
 
-        join_all(futures).await;
+        let catalogs = tokio::spawn({
+            let self_clone = self.clone();
+            async move {
+                self_clone.load_catalogs().await;
+            }
+        });
+
+        let models = tokio::spawn({
+            let self_clone = self.clone();
+            async move {
+                self_clone.load_models().await;
+            }
+        });
+
+        // Wait for all tasks to complete
+        let load_result = tokio::try_join!(task_history, results_cache, datasets, catalogs, models);
+
+        if let Err(err) = load_result {
+            tracing::error!("Could not start the Spice runtime: {err}");
```

Comment:
Should we report which subsystem failed to start?

Reply:
There is already logging at the inner layer for that. This is the error if the JoinHandle returns an error, which would be an internal Tokio issue - which is highly unlikely in practice.